SingularityExecutor/src/main/resources/logrotate.conf.hbs

```diff
@@ -14,15 +14,46 @@ notifempty
 
 {{#if shouldLogRotateLogFile }}
 {{{ logfile }}} {
-  dateformat {{{ rotateDateformat }}}{{#if logfileExtension}}.{{{ logfileExtension}}}{{/if}}
+  dateformat -{{{ rotateDateformat }}}{{#if logfileExtension}}.{{{ logfileExtension}}}{{/if}}
   olddir {{{ rotateDirectory }}}
+{{#if useFileAttributes}}
+  postrotate
+    NOW="$(($(date +%s%N)/1000000))"
+    LOGSTART=`getfattr --only-values -n user.logstart {{{ logfile }}}`
+    if [ "$LOGSTART" != "" ]; then
+      for filename in "{{{ rotateDirectory }}}/{{{ logfileName }}}-$(date +{{{ rotateDateformat }}})"*
+      do
+        setfattr -n user.logend -v "$NOW" {{{ rotateDirectory }}}/$filename
+        setfattr -n user.logstart -v "$LOGSTART" {{{ rotateDirectory }}}/$filename
+      done
+    fi
+    setfattr -n user.logstart -v "$NOW" {{{ logfile }}}
```

Comment:
Two things on that:
- I'm also setting this in the runner.sh script so a brand new log file would have it set.
- Right now in the S3Uploader I'm only adding the metadata if it is present on the file. The assumption would be that if we can't find a start time, we use the start time of the task itself, since it can be no earlier than that, and the same for end time. That's how the searching script currently works for all logs now